### PR TITLE
Standard library documentation continued.

### PR DESCRIPTION
**Pull request by yole**

### CODE REVIEW

File: core/builtins/native/kotlin/Primitives.kt [hunk 1]
```diff
@@ -25,65 +25,146 @@ package kotlin
...
+ * Returns zero if this value is equal to the specified other value, a negative number if its less than other, 
+ * or a positive number if its greater than other.
+ */
     public override fun compareTo(other: Byte): Int
+/**
+ * Compares this value with the specified value for order.
+ * Returns zero if this value is equal to the specified other value, a negative number if its less than other, 
+ * or a positive number if its greater than other.
+ */
     public fun compareTo(other: Char): Int
```
Comment (orangy) [inline core/builtins/native/kotlin/Primitives.kt]: Comparing byte with char for ordering probably requires more description about what it really means.

File: core/builtins/native/kotlin/Primitives.kt [hunk 2]
```diff
@@ -25,65 +25,146 @@ package kotlin
...
     public fun plus(other: Char): Int
+    /** Adds the two values. */
     public fun plus(other: Short): Int
+    /** Adds the two values. */
     public fun plus(other: Int): Int
+    /** Adds the two values. */
     public fun plus(other: Long): Long
+    /** Adds the two values. */
     public fun plus(other: Float): Float
+    /** Adds the two values. */
```
Comment (orangy) [inline core/builtins/native/kotlin/Primitives.kt]: "Adds the other value to this value."
like in `minus`

File: core/builtins/native/kotlin/Primitives.kt [hunk 3]
```diff
@@ -25,65 +25,146 @@ package kotlin
...
+    /** Subtracts the other value from this value. */
     public fun minus(other: Int): Int
+    /** Subtracts the other value from this value. */
     public fun minus(other: Long): Long
+    /** Subtracts the other value from this value. */
     public fun minus(other: Float): Float
+    /** Subtracts the other value from this value. */
     public fun minus(other: Double): Double
 
+    /** Multiplies the two values. */
```
Comment (orangy) [inline core/builtins/native/kotlin/Primitives.kt]: Multiplies this value by the other value.

File: core/builtins/native/kotlin/Primitives.kt [hunk 4]
```diff
@@ -100,65 +181,143 @@ public class Byte private () : Number, Comparable<Byte> {
...
     public fun inc(): Char
+    /** Decrements this value. */
     public fun dec(): Char
+    /** Returns this value. */
     public fun plus(): Int
+    /** Returns the negative of this value. */
     public fun minus(): Int
 
+     /** Creates a range from this value to the specified [other] value. */
     public fun rangeTo(other: Char): CharRange
```
Comment (orangy) [inline core/builtins/native/kotlin/Primitives.kt]: We don't have other overloads for rangeTo, like in other types?

File: core/builtins/native/kotlin/Primitives.kt [hunk 5]
```diff
@@ -100,65 +181,143 @@ public class Byte private () : Number, Comparable<Byte> {
...
 
+    /** Returns the value of this character as a `name`. */
```
Comment (orangy) [inline core/builtins/native/kotlin/Primitives.kt]: As a name?

File: core/builtins/src/kotlin/ExtensionFunctions.kt [hunk 6]
```diff
@@ -18,72 +18,118 @@
 
 package kotlin
 
+/** An extension function that takes 0 arguments. */
 public trait ExtensionFunction0<in T, out R> {
+    /** Invokes the function with the specified arguments. */
     public fun T.invoke(): R
 }
+/** An extension function that takes 1 arguments. */
```
Comment (orangy) [inline core/builtins/src/kotlin/ExtensionFunctions.kt]: "argument", no "s"

File: core/builtins/src/kotlin/ExtensionFunctions.kt [hunk 7]
```diff
@@ -18,72 +18,118 @@
 
 package kotlin
 
+/** An extension function that takes 0 arguments. */
 public trait ExtensionFunction0<in T, out R> {
+    /** Invokes the function with the specified arguments. */
```
Comment (orangy) [inline core/builtins/src/kotlin/ExtensionFunctions.kt]: There are no arguments, so just "Invokes the function"

File: core/builtins/src/kotlin/Functions.kt [hunk 8]
```diff
@@ -18,72 +18,118 @@
 
 package kotlin
 
+/** A function that takes 0 arguments. */
 public trait Function0<out R> {
+    /** Invokes the function with the specified arguments. */
```
Comment (orangy) [inline core/builtins/src/kotlin/Functions.kt]: no arguments

File: core/builtins/src/kotlin/Functions.kt [hunk 9]
```diff
@@ -18,72 +18,118 @@
...
     public fun invoke(): R
 }
+/** A function that takes 1 arguments. */
 public trait Function1<in P1, out R> {
+    /** Invokes the function with the specified arguments. */
     public fun invoke(p1: P1): R
 }
+/** A function that takes 2 arguments. */
```
Comment (orangy) [inline core/builtins/src/kotlin/Functions.kt]: plural use for 1

File: core/builtins/src/kotlin/Functions.kt [hunk 10]
```diff
@@ -18,72 +18,118 @@
 
 package kotlin
 
+/** A function that takes 0 arguments. */
 public trait Function0<out R> {
+    /** Invokes the function with the specified arguments. */
     public fun invoke(): R
 }
+/** A function that takes 1 arguments. */
```
Comment (orangy) [inline core/builtins/src/kotlin/Functions.kt]: plural use for 1

File: core/builtins/src/kotlin/Inline.kt [hunk 11]
```diff
@@ -55,6 +58,8 @@ public enum class InlineStrategy {
  * receiving function are allowed to use non-local control flow statements. Lambdas which are called from
  * a different execution context (for example, from an object contained in the receiving function)
  * are restricted to local control flow statements.
+ *
+ * @property value the inlining options selected for this lambda.
```
Comment (orangy) [inline core/builtins/src/kotlin/Inline.kt]: "this lambda" is unclear, may be "options for annotated functional parameter"?

File: libraries/stdlib/src/kotlin/collections/ArraysJVM.kt [hunk 12]
```diff
@@ -75,6 +75,9 @@ public fun ByteArray.toString(charset: String): String = String(this, charset)
  */
 public fun ByteArray.toString(charset: Charset): String = String(this, charset)
 
+/**
+ * Returnsn an array containing the elements of this collection.
```
Comment (orangy) [inline libraries/stdlib/src/kotlin/collections/ArraysJVM.kt]: typo "returnsn"

File: libraries/stdlib/src/kotlin/concurrent/Thread.kt [hunk 13]
```diff
@@ -2,36 +2,65 @@ package kotlin.concurrent
 
 import java.util.concurrent.*
 
+/**
+ * Return the current thread.
```
Comment (orangy) [inline libraries/stdlib/src/kotlin/concurrent/Thread.kt]: Returns

File: libraries/stdlib/src/kotlin/concurrent/Thread.kt [hunk 14]
```diff
@@ -53,15 +82,15 @@ public fun thread(start: Boolean = true, daemon: Boolean = false, contextClassLo
 
 /**
  * Allows you to use the executor as a function to
```
Comment (orangy) [inline libraries/stdlib/src/kotlin/concurrent/Thread.kt]: "Allows you" - not good for docs.

File: libraries/stdlib/src/kotlin/concurrent/Thread.kt [hunk 15]
```diff
@@ -53,15 +82,15 @@ public fun thread(start: Boolean = true, daemon: Boolean = false, contextClassLo
...
+ * execute the given block on the [Executor].
  */
 public fun Executor.invoke(action: () -> Unit) {
     execute(action)
 }
 
 /**
- * Allows you to use the executor as a function to
- * execute the given block on the [[Executor]].
+ * Allows you to use the executor service as a function to
```
Comment (orangy) [inline libraries/stdlib/src/kotlin/concurrent/Thread.kt]: Allows you.

File: libraries/stdlib/src/kotlin/concurrent/Timer.kt [hunk 16]
```diff
@@ -4,66 +4,121 @@ import java.util.Timer
...
+ * @param name the name to use for the thread which is running the timer.
+ * @param daemon if true, the thread is started as a daemon thread (the VM will exit when only daemon threads are running)
+ */
 public fun fixedRateTimer(name: String? = null, daemon: Boolean = false, startAt: Date, period: Long, action: TimerTask.() -> Unit): Timer {
     val timer = if (name == null) Timer(daemon) else Timer(name, daemon)
     timer.scheduleAtFixedRate(startAt, period, action)
     return timer
 }
 
+deprecated("Use the built-in SAM conversion instead")
```
Comment (orangy) [inline libraries/stdlib/src/kotlin/concurrent/Timer.kt]: Shouldn't we update usages in this file to invoke TimerTask constructor directly?

File: core/builtins/native/kotlin/Primitives.kt (hunk 4 above)
Comment (yole) [inline core/builtins/native/kotlin/Primitives.kt]: No, and this is intentional (doesn't make sense).

File: core/builtins/native/kotlin/Primitives.kt (hunk 5 above)
Comment (yole) [inline core/builtins/native/kotlin/Primitives.kt]: fixed

File: libraries/stdlib/src/kotlin/text/StringsJVM.kt [hunk 17]
```diff
@@ -7,127 +7,358 @@ import java.util.regex.MatchResult
...
 public fun String.endsWith(ch: Char): Boolean = (this as java.lang.String).endsWith(ch.toString())
 
 // "constructors" for String
 
+/**
+ * Converts the data from a portion of the specified array of bytes to characters using the specified character set
+ * and returns the conversion result as a string.
+ *
+ * @param offset the offset in the array of the data to be converted.
+ * @param length the number of bytes to be converted.
```
Comment (orangy) [inline libraries/stdlib/src/kotlin/text/StringsJVM.kt]: charsetName parameter missing

File: libraries/stdlib/src/kotlin/text/StringsJVM.kt [hunk 18]
```diff
@@ -7,127 +7,358 @@ import java.util.regex.MatchResult
...
+ */
 public fun String(bytes: ByteArray, offset: Int, length: Int, charsetName: String): String = java.lang.String(bytes, offset, length, charsetName) as String
 
+/**
+ * Converts the data from a portion of the specified array of bytes to characters using the specified character set
+ * and returns the conversion result as a string.
+ *
+ * @param offset the offset in the array of the data to be converted.
+ * @param length the number of bytes to be converted.
+ */
```
Comment (orangy) [inline libraries/stdlib/src/kotlin/text/StringsJVM.kt]: charset parameter missing

**Comment by yole** [comment]
Merged manually.

**Pull request closed without merging**